src/views/leads/Lead.js

```diff
@@ -296,23 +309,37 @@ const Lead = ({ brands, loggedUser }) => {
               </TableRow>
             </TableHead>
             <TableBody>
-              {myLeadListings.length > 0 ? (
+              {myLeadListings?.leads?.length > 0 ? (
                 <>
-                  {myLeadListings.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage).map(row => {
+                  {myLeadListings?.leads?.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage).map(row => {
                     return (
                       <TableRow hover role='checkbox' tabIndex={-1} key={row.code}>
```

Comment:
Are we fetching all the leads on the UI and then doing pagination on the UI ?
If yes, this is completely wrong.
the pagination should be on the backend.

Reply:
And adding pagination in the backend should be simple.
We just need to keep a state value `filter` as Object.
And set page, and size as keys in this object (including other filterS)
And just pass this object in the request to backend.